tests/ops/test_qubit_ops.py

```diff
@@ -941,6 +943,217 @@ def test_controlled_phase_shift_decomp(self, phi):
+        exp = DoubleExcitationMinus(phi)
+        assert np.allclose(res, exp)
+
+    @pytest.mark.parametrize("phi", [-0.1, 0.2, np.pi / 4])
+    def test_double_excitation_minus_generator(self, phi):
+        """Tests that the DoubleExcitationMinus operation calculates the correct generator"""
+        op = qml.DoubleExcitationMinus(phi, wires=[0, 1, 2, 3])
+        g, a = op.generator
+
+        res = expm(1j * a * g * phi)
+        exp = DoubleExcitationMinus(phi)
+
+        assert np.allclose(res, exp)
+
+    @pytest.mark.parametrize("excitation", [qml.DoubleExcitation, qml.DoubleExcitationPlus,
+                                            qml.DoubleExcitationMinus])
+    def test_autograd(self, excitation):
+        """Tests that operations are computed correctly using the
+        autograd interface"""
+
+        pytest.importorskip("autograd")
+
+        dev = qml.device('default.qubit.autograd', wires=4)
+        state = np.array([0, 0, 0, -1 / np.sqrt(2), 0, 0, 0, 0, 0, 0, 0, 0, 1 / np.sqrt(2), 0, 0,
+                          0])
+
+        @qml.qnode(dev)
+        def circuit(phi):
+            qml.PauliX(wires=0)
+            qml.PauliX(wires=1)
+            excitation(phi, wires=[0, 1, 2, 3])
+
+            return qml.state()
+
+        assert np.allclose(state, circuit(np.pi / 2))
+
+
+    @pytest.mark.parametrize("excitation", [qml.DoubleExcitation, qml.DoubleExcitationPlus,
+                                            qml.DoubleExcitationMinus])
+    def test_tf(self, excitation):
+        """Tests that operations are computed correctly using the
+        tensorflow interface"""
+
+        pytest.importorskip("tensorflow")
+
+        dev = qml.device('default.qubit.tf', wires=4)
+        state = np.array([0, 0, 0, -1 / np.sqrt(2), 0, 0, 0, 0, 0, 0, 0, 0, 1 / np.sqrt(2), 0, 0,
+                          0])
+
+        @qml.qnode(dev)
+        def circuit(phi):
+            qml.PauliX(wires=0)
+            qml.PauliX(wires=1)
+            excitation(phi, wires=[0, 1, 2, 3])
+
+            return qml.state()
+
+        assert np.allclose(state, circuit(np.pi / 2))
+
+    @pytest.mark.parametrize("excitation", [qml.DoubleExcitation, qml.DoubleExcitationPlus,
+                                            qml.DoubleExcitationMinus])
+    def test_jax(self, excitation):
+        """Tests that operations are computed correctly using the
+        jax interface"""
+
+        pytest.importorskip("jax")
+
+        dev = qml.device('default.qubit.jax', wires=4)
+        state = np.array([0, 0, 0, -1 / np.sqrt(2), 0, 0, 0, 0, 0, 0, 0, 0, 1 / np.sqrt(2), 0, 0,
+                          0])
+
+        @qml.qnode(dev)
+        def circuit(phi):
+            qml.PauliX(wires=0)
+            qml.PauliX(wires=1)
+            excitation(phi, wires=[0, 1, 2, 3])
+
+            return qml.state()
+
+        assert np.allclose(state, circuit(np.pi / 2))
+
+    @pytest.mark.parametrize(("excitation", "phi"), [(qml.DoubleExcitation, -0.1),
```

Comment:
Question here too: could we do 1) parametrizing for parameter-shift 2) testing parameter-shift for `default.qubit`?